zipline/algorithm.py

```diff
@@ -424,6 +424,9 @@ def _create_generator(self, sim_params, source_filter=None):
         processed by the zipline, and False for those that should be
         skipped.
         """
+        if sim_params is not None:
```

Comment:
It looks like they are being used here, https://github.com/quantopian/zipline/blob/jean-zipline-tests/zipline/algorithm.py#L454

The mixture usage of `self.sim_params` vs. `sim_params` is confusing in this method is confusing.